ibm_cloud_sdk_core/base_service.py

```diff
@@ -290,7 +290,8 @@ def prepare_request(self,
 
         if data and isinstance(data, dict):
             data = remove_null_values(data)
-            headers.update({'content-type': 'application/json'})
+            if headers.get('content-type') is None:
```

Comment:
Just curious... what if the content-type is something other than "application/json" (i.e. "application/octet-stream") but yet "data" is a dict?   It looks like we'd send the JSON stringified version of the dict in the request body, along with content-type="application/octet-stream".   Not 100% if we could get ourselves into this situation, but if so, we should probably decide what to do in that case.   For example, maybe we should have logic such that if "data" is a dict and the content-type is not None but also not a JSON-like value (includes json-patch, etc.) then we throw an error or a warning or something?